aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1499,7 +1499,7 @@ std::tuple<Tensor, Tensor, Tensor> _svd_helper_cuda(const Tensor& self, bool som
     }
 
     U_working_copy = same_stride_to(U_working_copy, self.options());
-    S_working_copy = same_stride_to(S_working_copy, self.options());
+    S_working_copy = same_stride_to(S_working_copy, S_working_copy.options().device(self.device()));
```

Comment:
should this bug fix go in 1.7?
cc. @nataliakliushkina

Reply:
The bug is that singular values for complex inputs would be returned with complex dtype? It's for you to decide @anjali411. Either way, there should be an svd test that would check returned type.

Reply:
@IvanYashchuk can you please update svd documentation to mention that svd supports complex matrices, and to mention the return types of S, U, V.

Reply:
Yes, that info will be added together with the fix for issue #45821.